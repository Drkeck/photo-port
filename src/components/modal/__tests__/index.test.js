import React from 'react'
import {render, cleanup, fireEvent} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from '..';

const currentPhoto = {
    name: 'Park bench',
    category: 'landscape',
    description: 'I am the description',
    index: 1
};

const mockToggleModel = jest.fn();

describe('Modal component', () => {
    it('renders', () => {
        render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModel}></Modal>)
    })

    it('Matches snapsot DOM node structure', () => {
        const { asFragment } = render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModel}></Modal>)
        expect(asFragment).toMatchSnapshot();
    })
})

describe('click event', () => {
    it('calls on close', () => {
        const { getByText } = render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModel}></Modal>)
        fireEvent.click(getByText('Close this Modal'));

        expect(mockToggleModel).toHaveBeenCalledTimes(1)
    })
})