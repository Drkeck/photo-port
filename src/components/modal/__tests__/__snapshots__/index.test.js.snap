// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Matches snapsot DOM node structure 1`] = `[Function]`;
